doc/kdump/kdump_Remote_SSH_HLD.md

```diff
@@ -0,0 +1,226 @@
+## Overview
+This document outlines the configuration and usage of the kdump remote feature with ssh for the SONiC.
+
+## Scope
+
+This document describes how to configure remote kdump feature in SONiC infrastructure.
+
+## Definitions/Abbreviations
+
+### Table 1: Abbreviations
+
+| **Term**     | **Meaning**            |
+|  ----------- | ---------------------- |
+| SSH          | Secure Shell           |
+| kdump        | Kernel Dump            |
+| NFS          | Network File System    |
+
+## Introduction
+Kdump, a built-in Linux kernel feature, generates and stores a crash dump file in the event of a kernel panic. Currently SONiC lacks the functionality of storing dump files on a remote server. It offloads the storage from switch by saving the crash reports to a separate designated server for offline analysis. This feature extends existing kdump feature by enabling remote dumps via ssh protocol, allowing you to transfer kernel crash data to a designated remote server.
+
+## Requirements Overview  <a name="requirements-overview"></a>
+
+### Functional Requirements  <a name="functional-requirements"></a>
+This section describes the SONiC requirements for kdump remote feature.
+
+At a high level the following should be supported:
+1. The kernel core dump files must be stored on the a remote ssh server.
+### Configuration and Management Requirements
+
+- CLI support for configuring remote kdump feature enable/disable via ssh.
+- CLI support for configuring username and hostname of ssh server (username@server_address).
+- CLI support for configuring SSH private key path for ssh server (SSH_private_Key_Path).
+- CLI support for displaying crededentials of ssh server.
+- CLI support for displaying state of kdump remote feature (enable/disable).
+### SSH Key Generation Requirement
+The system should authenticate with the remote server using SSH keys for secure access. 
+
+```
+admin@sonic: ssh-keygen
+```
+
+User be prompted to choose a location to save the key pair. By default, it saves the private key to ~/.ssh/id_rsa and the public key to ~/.ssh/id_rsa.pub.
+
+```
+admin@sonic: ssh-copy-id username@server_address
+```
+
+ This helps automate passwordless SSH logins by copying public key to authorized servers.
+
+## kdump Remote Architechture
+![alt text](./images/kdump-ssh.drawio.png)
+
+## Functional Description
+
+### Design Changes
+
+The SONiC kernel core dump remote functionality can be divided into two categories:
+
+1. Kernel core-dump generation service
+2. Storing Kernel core-dump files remotely
+
+Current SONiC lacks remote kernel dump functionality. To add this feature, consider enabling kdump for remote storage.
+
+We are suggesting modifying the exisitng SONiC configuration as following sonic-buildimage files.
+ 
+1. build_debian.sh (Addition)
+
+    - Required for kdump_remote_ssh_dump: Initialize network interfaces and enable DHCP upon kernel crash. 
+
+2. files/scripts/network_setup.sh (New Addition)
```

Comment:
Can you add some details for this new change? this is only for eth0 interface?

Reply:
Yes, this script initializes the eth0 interface and enables dhcp on it.